core/src/main/java/org/bitcoinj/coinjoin/CoinJoinClientSession.java

```diff
@@ -155,6 +158,13 @@ public int compare(CompactTallyItem o, CompactTallyItem t1) {
         return false;
     }
 
+    public void processTransaction(Transaction tx) {
+        log.info("processing tx of {}: {}", tx.getValue(mixingWallet), tx.getTxId());
+        if (collateralSessionMap.containsKey(tx.getTxId())) {
+            queueTransactionListeners(tx, collateralSessionMap.get(tx.getTxId()), CoinJoinTransactionType.MIXING_FEE);
+        }
+    }
+
```

Comment:
The main point of these `processTransaction` methods is to see if collateral TXes are spent.  If they are, we want to link them back to a Session Id so that it can be determined if the fee was part of the randomly paid fees of successful sessions or if there was a problem with our client and we were charged for bad behavior.  

Currently, DashJ doesn't keep track of this, but it is printed out in a coinjoin-report that can me looked at by the developers.

At this point all fees were charged to successful mixing sessions, which means we don't need to fix bugs related to session failures at this time.